ksql-engine/src/test/java/io/confluent/ksql/function/udaf/max/IntegerMaxKudafTest.java

```diff
@@ -43,4 +41,24 @@ public void shouldFindCorrectMax() {
     assertThat(8, equalTo(currentMax));
   }
 
+  @Test
+  public void shouldFindCorrectMaxForMerge() {
+    IntegerMaxKudaf integerMaxKudaf = getIntegerMaxKudaf();
+    Merger<String, Integer> merger = integerMaxKudaf.getMerger();
+    Integer mergeResult1 = merger.apply("Key", 10, 12);
+    assertThat(mergeResult1, equalTo(12));
+    Integer mergeResult2 = merger.apply("Key", 10, -12);
+    assertThat(mergeResult2, equalTo(10));
+    Integer mergeResult3 = merger.apply("Key", -10, 0);
+    assertThat(mergeResult3, equalTo(0));
+
+  }
+
+  private IntegerMaxKudaf getIntegerMaxKudaf() {
```

Comment:
Could you not just create the instance in `setUp`? Plus, is there a need to use `MaxAggFunctionFactory`? This is a unit test and should ideally only test the one class.

```
  private IntegerMaxKudaf udf;

   @Before
  private void setUp() {
     this.udf = new IntegerMaxKudaf(0);
  }
```